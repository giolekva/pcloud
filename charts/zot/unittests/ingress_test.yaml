suite: test ingress
templates:
  - ingress.yaml
tests:
  - it: should be empty if ingress is not enabled
    asserts:
      - hasDocuments:
          count: 0
  - it: should have apiVersion networking.k8s.io/v1 for k8s >= 1.19
    set:
      ingress.enabled: true
    capabilities:
      majorVersion: 1
      minorVersion: 19
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: Ingress
      - isAPIVersion:
          of: networking.k8s.io/v1
  - it: should have an ingressClassName for k8s >= 1.19
    set:
      ingress.enabled: true
      ingress.className: nginx-test
    capabilities:
      majorVersion: 1
      minorVersion: 19
    asserts:
      - hasDocuments:
          count: 1
      - equal:
          path: spec.ingressClassName
          value: nginx-test
  - it: should match snapshot of default values
    set:
      ingress.enabled: true
      ingress.className: nginx-test
      ingress.annotations:
        kubernetes.io/ingress.class: nginx
        kubernetes.io/tls-acme: "true"
      ingress.tls:
        - secretName: chart-example-tls
          hosts:
            - chart-example.local
    asserts:
      - matchSnapshot: { }
