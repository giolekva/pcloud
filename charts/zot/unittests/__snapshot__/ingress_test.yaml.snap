should match snapshot of default values:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      annotations:
        kubernetes.io/ingress.class: nginx
        kubernetes.io/tls-acme: "true"
      labels:
        app: zot
        release: RELEASE-NAME
      name: RELEASE-NAME-zot
    spec:
      ingressClassName: nginx-test
      rules:
        - host: chart-example.local
          http:
            paths:
              - backend:
                  service:
                    name: RELEASE-NAME-zot
                    port:
                      number: 5000
                path: /
                pathType: ImplementationSpecific
      tls:
        - hosts:
            - chart-example.local
          secretName: chart-example-tls
